src/__tests__/events.js

```diff
@@ -0,0 +1,287 @@
+      fireEvent[eventName](node)
+      expect(spy).toHaveBeenCalledTimes(1)
+    })
+  })
+})
+
+describe('UI Events', () => {
+  ;['scroll'].forEach(eventName => {
+    it(`fires ${eventName}`, () => {
+      const node = document.createElement('div')
+      const spy = jest.fn()
+      node.addEventListener(eventName.toLowerCase(), spy)
+      fireEvent[eventName](node)
+      expect(spy).toHaveBeenCalledTimes(1)
+    })
+  })
+})
+
+describe('Wheel Events', () => {
+  ;['wheel'].forEach(eventName => {
+    it(`fires ${eventName}`, () => {
+      const node = document.createElement('div')
+      const spy = jest.fn()
+      node.addEventListener(eventName.toLowerCase(), spy)
+      fireEvent[eventName](node)
+      expect(spy).toHaveBeenCalledTimes(1)
+    })
+  })
+})
+
+describe('Media Events', () => {
+  ;[
+    'abort',
+    'canPlay',
+    'canPlayThrough',
+    'durationChange',
+    'emptied',
+    'encrypted',
+    'ended',
+    'error',
+    'loadedData',
+    'loadedMetadata',
+    'loadStart',
+    'pause',
+    'play',
+    'playing',
+    'progress',
+    'rateChange',
+    'seeked',
+    'seeking',
+    'stalled',
+    'suspend',
+    'timeUpdate',
+    'volumeChange',
+    'waiting',
+  ].forEach(eventName => {
+    it(`fires ${eventName}`, () => {
+      const node = document.createElement('video')
+      const spy = jest.fn()
+      node.addEventListener(eventName.toLowerCase(), spy)
+      fireEvent[eventName](node)
+      expect(spy).toHaveBeenCalledTimes(1)
+    })
+  })
+})
+
+describe('Image Events', () => {
+  ;['load', 'error'].forEach(eventName => {
+    it(`fires ${eventName}`, () => {
+      const node = document.createElement('img')
+      const spy = jest.fn()
+      node.addEventListener(eventName.toLowerCase(), spy)
+      fireEvent[eventName](node)
+      expect(spy).toHaveBeenCalledTimes(1)
+    })
+  })
+})
+
+describe('Animation Events', () => {
+  ;['animationStart', 'animationEnd', 'animationIteration'].forEach(
+    eventName => {
+      it(`fires ${eventName}`, () => {
+        const node = document.createElement('div')
+        const spy = jest.fn()
+        node.addEventListener(eventName.toLowerCase(), spy)
+        fireEvent[eventName](node)
+        expect(spy).toHaveBeenCalledTimes(1)
+      })
+    },
+  )
+})
+
+describe('Transition Events', () => {
+  ;['transitionEnd'].forEach(eventName => {
+    it(`fires ${eventName}`, () => {
+      const node = document.createElement('div')
+      const spy = jest.fn()
+      node.addEventListener(eventName.toLowerCase(), spy)
+      fireEvent[eventName](node)
+      expect(spy).toHaveBeenCalledTimes(1)
+    })
+  })
+})
+
+describe('Alias Events', () => {
+  it(`fires doubleClick`, () => {
+    const node = document.createElement('button')
+    const spy = jest.fn()
+    node.addEventListener('dblclick', spy)
+    fireEvent.doubleClick(node)
+    expect(spy).toHaveBeenCalledTimes(1)
+  })
+})
+
+describe('React Events', () => {
```

Comment:
May be we need to have these test cases in `react-testing-library`? I'm just wondering because, anyway people are gonna use `dom-testing-library` for several frameworks (`angular`,`preact`,`vue` etc), do we have to add those frameworks testing here in future too?

Reply:
Also by moving it to `react-testing-library`, I guess people have good chance of looking at these test cases and understanding how to test events with `react`.

Reply:
Also moving it to react-testing-library will move the respective package dependencies that don't make sense in just the dom-testing-library

Reply:
Yes, also what we have created in `src/__tests__/helpers/react.js` is already part of `react-testing-library` (`render`,`unmount`).

Reply:
Personally I think it'd be better if we could avoid react in here and have a simple render method that creates DOM nodes manually. It doesn't need to be complicated.

Reply:
makes sense, i can open a PR to `react-testing-library` as well. there are a couple events we'll need to fallback to triggering synthetic events for (`change`, `select`, `mouseenter`, `mouseleave`) so I'll add that logic there

Reply:
Perfect. Thanks a bunch @jomaxx. This is so great!